src/Data/AbstractAugmented.php

```diff
@@ -44,20 +44,23 @@ public function select($keys = null)
 
     abstract public function keys();
 
+    public function getAugmentedMethodValue($method)
+    {
+        if ($this->methodExistsOnThisClass($method)) {
+            return $this->$method();
+        }
+
+        return $this->data->$method();
+    }
+
     public function get($handle): Value
     {
         $method = Str::camel($handle);
 
         if ($this->methodExistsOnThisClass($method)) {
-            $value = $this->$method();
-
-            return $value instanceof Value
-                ? $value
-                : new Value($value, $method, null, $this->data);
-        }
-
-        if (method_exists($this->data, $method) && collect($this->keys())->contains(Str::snake($handle))) {
-            return $this->wrapValue($this->data->$method(), $handle);
+            return $this->wrapInvokable($method, true, $this, $handle);
+        } elseif (method_exists($this->data, $method) && collect($this->keys())->contains(Str::snake($handle))) {
```

Comment:
there's a return in the `if`, why the `else if`?

Reply:
☕deficiency